mesonbuild/environment.py

```diff
@@ -736,6 +736,8 @@ def mfilestr2key(self, machine_file_string: str, section: T.Optional[str], secti
         if section_subproject:
             key = key.evolve(subproject=section_subproject)
         if machine == MachineChoice.BUILD:
+            if key.machine == MachineChoice.BUILD:
+                mlog.deprecation('Setting build machine options in the native file does not need the "build." prefix', once=True)
```

Comment:
Commit message says "and remove the assert" but it appears you forgot to add that part?

Reply:
Weird, I certainly did that before.